ownCloud/Settings/Passcode/PasscodeViewController.swift

```diff
@@ -0,0 +1,306 @@
+
+// MARK: - Interface view mode
+enum PasscodeInterfaceMode {
+    case addPasscodeFirstStep
+    case addPasscodeSecondStep
+    case unlockPasscode
+    case unlockPasscodeError
+    case deletePasscode
+    case deletePasscodeError
+    case addPasscodeFirstSetpAfterErrorOnSecond
+}
+
+class PasscodeViewController: UIViewController, Themeable {
+
+    // MARK: - Handler
+    var handler: PasscodeHandler?
+
+    // MARK: - Overlay
+    var passcodeFromFirstStep: String?
+    var passcodeMode: PasscodeInterfaceMode?
+    var hiddenOverlay: Bool?
+
+    // MARK: - Overlay view
+    @IBOutlet var overlayView: UIView!
+    @IBOutlet var logoTVGView : VectorImageView!
+
+    // MARK: - Messages and input text
+    @IBOutlet weak var messageLabel: UILabel?
+    @IBOutlet weak var errorMessageLabel: UILabel?
+    @IBOutlet weak var passcodeValueTextField: UITextField?
+
+    // MARK: - Buttons
+    @IBOutlet weak var number0Button: ThemeButton?
+    @IBOutlet weak var number1Button: ThemeButton?
+    @IBOutlet weak var number2Button: ThemeButton?
+    @IBOutlet weak var number3Button: ThemeButton?
+    @IBOutlet weak var number4Button: ThemeButton?
+    @IBOutlet weak var number5Button: ThemeButton?
+    @IBOutlet weak var number6Button: ThemeButton?
+    @IBOutlet weak var number7Button: ThemeButton?
+    @IBOutlet weak var number8Button: ThemeButton?
+    @IBOutlet weak var number9Button: ThemeButton?
+
+    @IBOutlet weak var cancelButton: ThemeButton?
+
+    // MARK: - Initalization view
+    init(mode: PasscodeInterfaceMode, hiddenOverlay: Bool?, handler: PasscodeHandler?) {
+        super.init(nibName: "PasscodeViewController", bundle: nil)
+        self.passcodeFromFirstStep = nil
+        self.passcodeMode = mode
+        self.hiddenOverlay = hiddenOverlay
+        self.handler = handler
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        Theme.shared.register(client: self)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super .viewDidAppear(animated)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        Theme.shared.unregister(client: self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.loadUI()
+        UIDevice.current.setValue(UIInterfaceOrientation.portrait.rawValue, forKey: "orientation")
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    // MARK: - Rotation Control
+
+    override var shouldAutorotate: Bool {
+        return false
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return .portrait
+    }
+
+    // MARK: - User Interface
+
+    private func loadUI() {
```

Comment:
I setted it on a separate method in order to have all the things relative about the interface separated because maybe in the future or in a different ViewController we need to have many things on the `viewDidLoad` and doing this we have it separated.
But if you want I can put it on the `viewDidLoad`

Reply:
Please move it to viewDidLoad.

If a subclass needs to do more in the future, it can override that viewDidLoad, call super.viewDidLoad() and add its customizations.